### PR TITLE
feat(breakingchange): :sparkles: add detectBreakingChange option

### PR DESCRIPTION
**Pull request by yi-Xu-0100**
Ref #39

### CODE REVIEW

File: src/lib/conventional-commits.ts [hunk 1]
```diff
@@ -182,7 +183,17 @@ export default function createConventionalCommits() {
       output.appendLine(
         `commitMessage: ${JSON.stringify(commitMessage, null, 2)}`,
       );
-      const message = serialize(commitMessage);
+
+      let message = serialize(commitMessage);
+      const detectBreakingChange = configuration.get<boolean>(
+        'detectBreakingChange',
+      );
+      if (detectBreakingChange && /^BREAKING( |-)CHANGE: /gm.test(message)) {
```
Comment (vivaxy) [inline src/lib/conventional-commits.ts:191]: It's better to limit the matching within `footer` section.

File: src/lib/conventional-commits.ts [hunk 2]
```diff
@@ -182,7 +183,17 @@ export default function createConventionalCommits() {
...
+        output.appendLine(
+          'detectBreakingChange: Found `BREAKING CHANGE` in message.',
+        );
+        message = message.replace(': ', '!: ');
```
Comment (vivaxy) [inline src/lib/conventional-commits.ts:195]: 😦It looks like a patch, not a designed feature.

It's better not to replace the string, but to update the serialization function.

**Review by vivaxy — CHANGES_REQUESTED** [review]
🤩Thank you for your contribution.

I think we should redesign the `serialize` function(https://github.com/vivaxy/vscode-conventional-commits/blob/master/src/lib/commit-message.ts#L102) to implement the feature.

**Comment by yi-Xu-0100** [comment]
Ok, I will try to figure out the workflow about `serialize`. 😀

**Pull request closed without merging**